geomesa-memory/src/main/scala/org/locationtech/geomesa/memory/cqengine/utils/CQIndexingOptions.scala

```diff
@@ -0,0 +1,139 @@
+
+    // Add Geometry index on default geometry first.
+    addGeoIndex(sft.getGeometryDescriptor, cqcache)
+
+    // Add fid index
+    // TODO: should we always build this or expose building this as an option?
+    addFidIndex(cqcache)
+
+    // Add other indexes
+    sft.getAttributeDescriptors.foreach {
+      addIndex(_, cqcache)
+    }
+
+    cqcache
+  }
+
+  def addIndex(ad: AttributeDescriptor, coll: IndexedCollection[SimpleFeature]): Unit = {
+    getCQIndexType(ad) match {
+      case CQIndexType.DEFAULT   =>
+        ad.getType.getBinding match {
+          // Comparable fields should have a Navigable Index
+          case c if
+            classOf[java.lang.Integer].isAssignableFrom(c) ||
+            classOf[java.lang.Long].isAssignableFrom(c) ||
+            classOf[java.lang.Float].isAssignableFrom(c) ||
+            classOf[java.lang.Double].isAssignableFrom(c) ||
+            classOf[java.util.Date].isAssignableFrom(c)            => addNavigableIndex(ad, coll)
+          case c if classOf[java.lang.String].isAssignableFrom(c)  => addRadixIndex(ad, coll)
+          case c if classOf[Geometry].isAssignableFrom(c)          => addGeoIndex(ad, coll)
+          case c if classOf[UUID].isAssignableFrom(c)              => addUniqueIndex(ad, coll)
+          // TODO: Decide how boolean fields should be indexed
+          case c if classOf[java.lang.Boolean].isAssignableFrom(c) => addHashIndex(ad, coll)
+        }
+
+      case CQIndexType.NAVIGABLE => addNavigableIndex(ad, coll)
+      case CQIndexType.RADIX     => addRadixIndex(ad, coll)
+      case CQIndexType.UNIQUE    => addUniqueIndex(ad, coll)
+      case CQIndexType.HASH      => addHashIndex(ad, coll)
+      case CQIndexType.NONE      => // NO-OP
+    }
+  }
+
+  def addFidIndex(coll: IndexedCollection[SimpleFeature]): Unit = {
+    val attribute = SFTAttributes.fidAttribute
+    coll.addIndex(HashIndex.onAttribute(attribute))
+  }
+
+  def addGeoIndex(ad: AttributeDescriptor, coll: IndexedCollection[SimpleFeature]): Unit = {
+    // TODO: Add logic to allow for the geo-index to be disabled?  (Low priority)
+    val defaultGeom: Attribute[SimpleFeature, Geometry] =
+      new SimpleFeatureAttribute(classOf[Geometry], ad.getLocalName)
+    coll.addIndex(GeoIndex.onAttribute(defaultGeom))
+  }
+
+  def addNavigableIndex(ad: AttributeDescriptor, coll: IndexedCollection[SimpleFeature]): Unit = {
+    val binding = ad.getType.getBinding
+    binding match {
+      case c if classOf[java.lang.Integer].isAssignableFrom(c) => BuildIntNavIndex(ad, coll)
+      case c if classOf[java.lang.Long   ].isAssignableFrom(c) => BuildLongNavIndex(ad, coll)
+      case c if classOf[java.lang.Float  ].isAssignableFrom(c) => BuildFloatNavIndex(ad, coll)
+      case c if classOf[java.lang.Double ].isAssignableFrom(c) => BuildDoubleNavIndex(ad, coll)
+      case c if classOf[java.util.Date   ].isAssignableFrom(c) => BuildDateNavIndex(ad, coll)
+    }
```

Comment:
It might be worth adding a case _ => which logs a warning.  (Or throws an exception.  I'm feeling nice...)